architecture/core.rst

```diff
@@ -157,7 +157,7 @@ with IRQ and FIQ masked. SMCs are categorised in two flavors: **fast** and
     participant "Secure Monitor" as smon
     participant "OP-TEE OS entry" as entry
     participant "OP-TEE OS" as optee
-    == IRQ and FIQ unmasked ==
+    == IRQ and FIQ masked ==
```

Comment:
IRQs/FIQs maybe unmask when non-secure invokes OP-TEE. I think a better fix would be:
```patch
     participant "OP-TEE OS entry" as entry
     participant "OP-TEE OS" as optee
     == IRQ and FIQ unmasked ==
     nwd -> smon : smc: TEE_FUNC_INVOKE
+    == IRQ and FIQ masked ==
     smon -> smon : Save non-secure context
     smon -> smon : Restore secure context

Reply:
Would it be more accurate to leave it undefined in the diagram instead, and remove the `== IRQ and FIQ unmasked ==` line in your snippet? Or do we want to keep it mentioned as the 'worst case scenario'?

Reply:
It could but then at sequence end below, we would need to also mention 'undefined'.
To make it simpler here and consistent with the other sequences below, I think using "unmasked" state at entry/exit is better. Feel free to propose anything you think that could be more accurate, but still simple to read.

Reply:
I think your initial approach works well too. I updated the PR to reflect that.